plugin/net/cni.go

```diff
@@ -183,16 +183,21 @@ func assignBridgeIP(bridgeName string, ipnet net.IPNet) error {
 	return nil
 }
 
+// As of CNI 0.5 spec:
+//   "Plugins should generally complete a DEL action without error even if some resources are missing"
+// this method should therefore return nil in most, if not all, cases.
 func (c *CNIPlugin) CmdDel(args *skel.CmdArgs) error {
 	conf, err := loadNetConf(args.StdinData)
 	if err != nil {
-		return err
+		logToKubelet(err)
+		return nil
 	}
 
 	// As of CNI 0.3 spec, runtimes can send blank if they just want the address deallocated
 	if args.Netns != "" {
-		if _, err = weavenet.WithNetNS(args.Netns, "del-iface", args.IfName); err != nil {
-			return fmt.Errorf("error removing interface %q: %s", args.IfName, err)
+		if _, err = weavenet.WithNetNS(args.Netns, "del-iface", args.IfName); err != nil && !os.IsNotExist(err) {
```

Comment:
I thought it would work given:
- `WithNetNS` wraps around the error returned by `exec.Command` and 
- `IsNotExist` extracts the underlying error:
``` 
func isNotExist(err error) bool {
    err = underlyingError(err)
    return err == syscall.ENOENT || err == ErrNotExist
}
```